examples/08_maps/main.rs

```diff
@@ -0,0 +1,111 @@
+extern crate amethyst;
+extern crate cgmath;
+extern crate obj;
+extern crate yaml_rust;
+
+use std::sync::{Mutex, Arc};
+
+use amethyst::config::Element;
+use amethyst::context::Context;
+use amethyst::context::asset_manager::{Mesh, Texture};
+use amethyst::context::map_manager::load_map;
+use amethyst::ecs::{Join, Processor, RunArg, World};
+use amethyst::engine::{Application, Config, State, Trans};
+use amethyst::processors::rendering::{Camera, Light, Projection, RenderingProcessor, Renderable};
+use amethyst::processors::transform::{Child, Init, LocalTransform, Transform, TransformProcessor};
+
+struct MapsDemo;
+
+// Set up Processor for demo
+struct MapsDemoProcessor;
+
+unsafe impl Sync for MapsDemoProcessor {  }
+
+impl Processor<Arc<Mutex<Context>>> for MapsDemoProcessor {
+    fn run(&mut self, arg: RunArg, ctx: Arc<Mutex<Context>>) {
+
+        // Get all needed component storages and resources
+        let ctx = ctx.lock().unwrap();
+        let (
+            mut locals,
+        ) = arg.fetch(|w| (
+            w.write::<LocalTransform>(),
+        ));
+    }
+}
+
+impl State for MapsDemo {
+    fn on_start(&mut self, ctx: &mut Context, world: &mut World) {
+        let (w, h) = ctx.renderer.get_dimensions().unwrap();
+        let aspect = w as f32 / h as f32;
+
+        // Get a Perspective projection
+        let projection = Projection::Perspective {
+            fov: 60.0,
+            aspect: aspect,
+            near: 1.0,
+            far: 100.0,
+        };
+
+        world.add_resource::<Projection>(projection.clone());
+
+        // Create a camera entity
+        let eye =    [0., -50., 0.];
+        let target = [0.,  0., 0.];
+        let up =     [0.,  0., 1.];
+        let mut camera = Camera::new(projection, eye, target, up);
+        camera.activate();
+        world.create_now()
+            .with(camera)
+            .build();
+
+        // Set up world
+        ctx.asset_manager.register_asset::<Mesh>();
+        ctx.asset_manager.register_asset::<Texture>();
+
+        let result = load_map::<Renderable, Light, LocalTransform, Transform>(ctx, world, "examples/08_maps/resources/maps/demo.yml");
```

Comment:
I got this to a basic working example, but I imagine it would work better as something like this:

```rust
ctx.map_manager.register_asset::<Renderable>();
ctx.map_manager.register_asset::<Light>();
ctx.map_manager.register_asset::<LocalTransform>();
ctx.map_manager.register_asset::<Transform>();
let result = ctx.map_manager.load_file("examples/08_maps/resources/maps/demo.yml", ctx, world);
```